### PR TITLE
jms: add queue browser

### PR DESCRIPTION
**Pull request by dwickern**
Exposes the JMS queue browser API which iterates messages on the queue without consuming them.

### CODE REVIEW

**Comment by dwickern** [comment]
The failing test is the same transient failure as #708

**Comment by ennru** [comment]
Can you add a section to the documentation in `jms.md` as well, please

File: jms/src/main/scala/akka/stream/alpakka/jms/JmsBrowseStage.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    new GraphStageLogic(shape) with OutHandler {
+      setHandler(out, this)
+
+      var connection: Connection = _
+      var session: Session = _
+      var browser: QueueBrowser = _
+      var messages: JEnumeration[Message] = _
+
+      override def preStart(): Unit = {
+        val queue = settings.destination.getOrElse { throw new IllegalArgumentException("Destination is missing") }
```
Comment (johanandren) [inline jms/src/main/scala/akka/stream/alpakka/jms/JmsBrowseStage.scala]: Check this on construction instead of materialization to fail as fast as possible

**Comment by ennru** [comment]
Thank you for your contribution!

**Pull request merged**